### ISSUE TITLE
HttpRequest::url_for for a named route with no variables does not include the path

### ISSUE THREAD

**Issue by Mange** [actix/actix-web#265]
I have an app with the following resources:

```rust
            .resource("/tasks", |r| {
                r.name("all_tasks");
                r.method(http::Method::GET).with(resources::tasks::index)
            })
            .resource("/tasks/{uuid}", |r| {
                r.name("show_task");
                r.method(http::Method::GET).with(resources::tasks::show)
            })
```

If I then generate URLs like this:

```rust
let no_params: [&str; 0] = [];
let url1 = req.url_for("all_tasks", &no_params)?.into_string();
let url2 = req.url_for("show_task", &[":task_uid"])?.into_string();
```

Then the returned URLs looks like this:

```rust
"http://localhost:8088/"  // (url1, all_tasks)
"http://localhost:8088/tasks/:task_uuid" // (url2, show_task)
```

It looks like when there are no parameters, then the path is not constructed in the call. I've read the source code but I could not figure out why this would happen.

Did I do something wrong here? Is `url_for` only for resources with variables? Is there some other way to generate the full URLs for a request?

```
› rg "\"actix[^ ]* \d" Cargo.lock
6: "actix_derive 0.2.0 (registry+https://github.com/rust-lang/crates.io-index)",
28: "actix 0.5.7 (registry+https://github.com/rust-lang/crates.io-index)",
1119: "actix-web 0.6.10 (registry+https://github.com/rust-lang/crates.io-index)",
```

**Comment by fafhrd91** [comment]
might be bug. let me check

**Comment by fafhrd91** [comment]
fixed in master and 0.6 branches

**Comment by Mange** [comment]
Thank you ♥️
